Lilu/Sources/kern_devinfo.cpp

```diff
@@ -209,6 +209,14 @@ void DeviceInfo::grabDevicesFromPciRoot(IORegistryEntry *pciRoot) {
 				DBGLOG("dev", "found IGPU device %s", safeString(name));
 				videoBuiltin = obj;
 				requestedExternalSwitchOff |= videoBuiltin->getProperty(RequestedExternalSwitchOffName) != nullptr;
+			} else if (vendor == WIOKit::VendorID::ATIAMD && (code == WIOKit::ClassCode::DisplayController || code == WIOKit::ClassCode::VGAController)) {
+				uint32_t dev = 0;
+				WIOKit::getOSDataValue(obj, "device-id", dev);
+				dev = (dev & 0xFF00);
+				if ((dev == 0x1600 || dev == 0x1500) || (dev == 0x9800 || dev == 0x1300) || (dev == 0x9900 || dev == 0x9600)) {
```

Comment:
If we end up having this code, these numbers must become named constants, so that it is clear which id belongs to which family.

Reply:
As for this, the generic &ing makes the Device ID range more generic so there isn’t a million cases.

0x1500 is commonly associated with AMD’s Raven line, 0x1600 catches devices from Renoir, Van Gogh and Rembrandt

0x1300 catches Kaveri (Spectre) and a certain Granite Ridge iGPU (0x13C0)

0x9800 catches Carrizo, Mullins (Godavari), Kabini (Kalindi) and Stoney

0x9600 and 0x9900 catch some of the legacy TeraScale iGPUs such as Trinity and Richland iirc

Reply:
This information must be in the code, not in the comments to the review. Unnamed numeric literals are plain bad, make them named constants.

Reply:
was going to do this already, got caught up in other things
pushed out the changes now